### PR TITLE
Prism bukkit 170 - Fix Untracked item movement issue 170

### PR DESCRIPTION
**Pull request by Narimm**
Fixes the Item bugs that prevent logging of single left click movements

### CODE REVIEW

File: src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java [hunk 1]
```diff
@@ -152,42 +149,65 @@ public void onInventoryClick(final InventoryClickEvent event) {
...
-                            int amount = Math.min(slotItem.getType().getMaxStackSize(),
-                                    slotItem.getAmount() + heldItem.getAmount()) - slotItem.getAmount();
-
-                            if (amount > 0) {
-                                RecordingQueue.addToQueue(ActionFactory.createItemStack(INSERT, slotItem, amount, slot,
-                                        null, containerLoc, player));
+                        int amount = 0;
+                        int maxStack = heldItem.getMaxStackSize();
+                        if (slotItem.getType() == Material.AIR) {
+                            if (heldItem.getAmount() <= maxStack) {
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java:169]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948&bid=18147497)

File: src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java [hunk 2]
```diff
@@ -336,4 +348,18 @@ public void onInventoryClick(final InventoryClickEvent event) {
                 // What the hell did you do
         }
     }
+
+    private int recordDeductTransfer(String act, int size, int amount, ItemStack heldItem, Location containerLoc,
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java:352]: Codacy Issue found: [Avoid reassigning parameters such as 'amount'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948&bid=18147497)

File: src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java (hunk 1 above)
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java:169]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)

File: src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java (hunk 2 above)
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java:352]: Codacy Issue found: [Avoid reassigning parameters such as 'amount'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948&bid=18147497)

File: src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java [hunk 3]
```diff
@@ -152,42 +149,63 @@ public void onInventoryClick(final InventoryClickEvent event) {
...
+                            Prism.debug("ACTION: " + event.getAction().name());
                         }
                     } else {
-                        if (slotItem.getType() == Material.AIR || slotItem.equals(heldItem)) {
-                            int amount = Math.min(slotItem.getType().getMaxStackSize(),
-                                    slotItem.getAmount() + heldItem.getAmount()) - slotItem.getAmount();
+                        int amount = 0;
+                        int maxStack = heldItem.getMaxStackSize();
+                        if (slotItem.getType() == Material.AIR && heldItem.getAmount() <= maxStack) {
+                                amount = heldItem.getAmount();
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java:169]: Codacy Issue found: ['if' child has incorrect indentation level 32, expected level should be 28.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948&bid=18147497)

**Review by addstar — DISMISSED** [review]

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948&bid=18147497)

**Comment by Narimm** [comment]
Closes #170

**Pull request merged**